utilities/helper-functions.metta

```diff
@@ -480,3 +480,82 @@
     )
   )
 )
+
+
+
+; below are helper functions for DELETE AND and OR subtree handlers 
+
+
+; takes a list of trees  and a nodeType returns a new tree excluding all the trees with the given NodeType (AND | OR | NOT | LITERAL)
+(: all-except-target (-> TreeList NodeType TreeList))
+(= (all-except-target $treeList $nodeType) (
+    case $treeList 
```

Comment:
try to use pattern matching when ever possible instead of cases

Reply:
I have reimplemented the functions with the new tree structure and less usage of cases.